support-frontend/assets/components/subscriptionCheckouts/address/postcodeFinder.tsx

```diff
@@ -9,123 +8,148 @@ import {
 	TextInput,
 } from '@guardian/source-react-components';
 import React from 'react';
-import type { PostcodeFinderState } from 'components/subscriptionCheckouts/address/postcodeFinderStore';
 import type { PostcodeFinderResult } from 'components/subscriptionCheckouts/address/postcodeLookup';
 
-const root = css`
-	display: flex;
-	justify-content: flex-start;
-	margin-bottom: ${space[6]}px;
-`;
-
-const inputStyles = css`
-	margin-right: ${space[3]}px;
-`;
-
-const buttonStyles = css`
-	align-self: flex-end;
-`;
-
-// Types
-type PostcodeFinderProps = PostcodeFinderState & {
-	id: string;
-	onPostcodeUpdate: (newPostcode: string) => void;
-	onAddressUpdate: (result: PostcodeFinderResult) => void;
-	setPostcode: (postcode: string) => void;
-	fetchResults: (postcode?: string) => void;
-};
+// ---- Component ---- //
 
-// Helpers
-function InputWithButton({
-	onClick,
-	isLoading,
-	...props
-}: TextInputProps & {
-	onClick: () => void;
+interface PostCodeFinderProps {
+	postcode: string;
 	isLoading: boolean;
-}) {
-	return (
-		<div css={root}>
-			<div>
-				<TextInput
-					{...props}
-					onKeyPress={(e) => {
-						if (e.key && e.key === 'Enter') {
-							e.preventDefault();
-							onClick();
-						}
-					}}
-					css={inputStyles}
-					name="postcode"
-					width={10}
-				/>
-			</div>
-
-			{!isLoading && (
-				<ThemeProvider theme={buttonThemeReaderRevenueBrandAlt}>
-					<Button
-						priority="tertiary"
-						css={buttonStyles}
-						type="button"
-						onClick={onClick}
-					>
-						Find address
-					</Button>
-				</ThemeProvider>
-			)}
-		</div>
-	);
+	results?: PostcodeFinderResult[];
+	error?: string;
+	onPostcodeUpdate: (postcode: string) => void;
+	onPostcodeError: (error: string) => void;
+	onFindAddress: (postcode: string) => void;
+	onAddressSelected: (result: PostcodeFinderResult) => void;
 }
 
 export function PostcodeFinder({
-	id,
 	postcode,
-	results,
 	isLoading,
-	setPostcode,
-	fetchResults,
+	results,
 	error,
 	onPostcodeUpdate,
-	onAddressUpdate,
-}: PostcodeFinderProps): JSX.Element {
+	onPostcodeError,
+	onFindAddress,
+	onAddressSelected,
+}: PostCodeFinderProps): JSX.Element {
+	function handleSubmit() {
+		if (isLoading) {
+			return;
+		}
+
+		if (postcode.trim() === '') {
+			onPostcodeError('Please enter a postcode');
+			return;
+		}
+
+		onFindAddress(postcode);
+	}
```

Comment:
This behaviour is slightly different than before. First, we now do an `isLoading` check to prevent kicking off a second search whilst the first is loading (even though the button isn't displayed, you can still hit enter in the text field to kick off the search). 

Second, we now do the error check inside of this submit handler, instead of inside of the `onFindAddress` callback - this was to keep the async thunk code as simple as possible